mantidproject/mantid
# Abins: fix q-dependence of estimated high-order terms

**Description of work.**

High-order phonon contributions are estimated by repeated convolution of the fundamental spectrum against the highest calculated order. (As this may be the first order, we refer to the method as 'autoconvolution'. But strictly speaking only the first step is actually "auto"...)

In Abins energy and q are related by the instrument/angle geometry, so the q^2 dependence of the spectrum is "baked in" to the calculation over energies at each angle. However, as high order spectra should scale with (q^2n) (where q is the quantum order), it is not correct to rely on the q information in the fundamental spectrum. Instead, the approach should be:

- Compute the spectra at a fixed q (1 recip. angstrom is convenient...)

- Apply the q^2n terms after q-independent line shapes are determined by convolution

Here we do this slightly inefficiently to fit the existing Abins toolchain (further refactors to be considered...) and

- Compute the order-1 spectrum as a function of q (with and without the order-1 Debye-Waller factor) and, if desired, the order-2 spectrum.

- Remove the q-dependence from the non-DW-corrected spectra by division

- Use the fixed-q order-1 spectrum as the basis for auto-convolution

- Apply (q^2n / n!) intensity correction to each kernel

- Apply q-dependent Debye-Waller factors exp(-tr(A) q^2/3) to all orders

- Replace order-1 data with the previously-computed spectrum (including a mode-dependent Debye-Waller term.)

This replaces an implementation in which the n! correction was neglected and convolution kernels were normalised to 1. It is believed that some other implementations of this method perform convolution
with normalised kernels; however it seems more convenient to leave the (tr(B)/3) term in the convolution than to remove it and later apply (tr(B)/3)^n. Perhaps this comes at a cost in precision?

### Work-in-progress checklist
- [x] Update sdata autoconvolution test, consider removing normalised mode
- [x] Check scaling is consistent when multiple k-points are included
- [x] Check results broadly agree with other implementations
- [ ] ~~Sort out a bug: Abins complains when calculating order-1 (and convolving the rest) if order-2 data already exists~~ **Opened issue #33598**

**To test:**


Abins outputs that _don't_ involve convolution are quite well tested already.
Results with convolution have changed; currently there is no test for results with convolution. A wider range of tests should be added for the Abins powder calculations - but they would not prove much about this PR so I suggest it is a separate piece of work.
Until then, manual testing will show the improvement: run Abins with and without the Autoconvolution option ticked. Both runs should work, and the former should have a nice spectrum of overtones, accumulating into a broadly flat shelf. Examples are shown in the Issue discussion below.

This will change user results (for the better!) and requires a release note. The release note has been written, by may need moving/adapting to the new system?



---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
